packages/opentelemetry-core/src/context/propagation/B3MultiPropagator.ts

```diff
@@ -95,11 +107,24 @@ function getTraceFlags(
 export class B3MultiPropagator implements TextMapPropagator {
   inject(context: Context, carrier: unknown, setter: TextMapSetter) {
     const spanContext = getParentSpanContext(context);
-    if (!spanContext || !isSpanContextValid(spanContext)) return;
+    const parentSpanId = context.getValue(B3_PARENT_SPAN_ID_KEY) as
```

Comment:
This is the parent span id of the extracted request, which is incorrect unless no other in-process spans have been created. Unfortunately, I don't see a way to propagate the parent span id for in-process spans using OpenTelemetry. While `parentSpanId` is available on ReadableSpan in the export pipeline, it's not part of the Span API. 

For B3, the parentSpanId is propagated so that both sides of a request can share the same span id. OpenTelemetry does not support span id sharing. By omitting parentSpanId, a downstream service should generate a new span id for the inbound request.

See the b3 spec (https://github.com/openzipkin/b3-propagation):

> Some propagation formats look similar to B3, but don't propagate a field named parent. Instead, they propagate a span ID field which serves the same purpose as ParentSpanId. Unlike B3, these systems use a different span ID for the client and server side of an RPC. When a server reads headers like this, it is expected to provision a new span ID for itself, and use the one it extracted as its parent.

...

> In both B3 and the above example, incoming headers contain the parent's span ID, and three IDs (trace, parent, span) end up in the trace context. The difference is that B3 uses the same span ID for the client and server side of an RPC, where the latter does not.

 See also this spec PR: https://github.com/open-telemetry/opentelemetry-specification/pull/1045

Reply:
I just saw your comment here: https://github.com/open-telemetry/opentelemetry-specification/pull/1045#issuecomment-714197375
Thanks for taking this into consideration :) I will close this PR, and keep following the progress with B3.
And thanks for taking the time to explain why `parentSpanId` propagation was intentionally removed.